lib/config/configitem.cpp

```diff
@@ -478,7 +476,11 @@ bool ConfigItem::CommitNewItems(const ActivationContext::Ptr& context, WorkQueue
 					return;
 				}
 
+				itemsCount++;
 				committed_items++;
+
+				std::unique_lock<std::mutex> lock(m_Mutex);
+				newItems.emplace_back(item);
```

Comment:
`m_Mutex` protects (static) members of `ConfigItem` which is completely unrelated to the use here. I'd use a separate mutex here (can be a local variable as it's only needed for synchronization within this one function here).